src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let currentState = Object.assign({}, state);
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'clear') {
+      currentState = {};
+      history.push(currentState);
```

Comment:
[DONT REPEAT YOURSELF] - You are pushing the currentState to the history array in every case. This action should be performed once after the switch statement to avoid repetition.